featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -118,6 +118,10 @@ def run(self, instance_ids):
 
         df.index.name = self.entityset[self.feature_set.target_eid].index
         column_list = []
+
+        # Order by instance_ids
+        df = df.reindex(list(pd.unique(instance_ids)))
```

Comment:
I am turning this into a `list` because `instance_ids` may be `pd.Categorical`, and `pd.unique` not only removes duplicate values but also changes the categories. The groups then can't be concatenated later because the categories are different.
```
>>> import pandas as pd
>>> cats = pd.Categorical([1, 2, 3, 4])
>>> cats
[1, 2, 3, 4]
Categories (4, int64): [1, 2, 3, 4]
>>> sub_cats = cats[:2]
>>> sub_cats
[1, 2]
Categories (4, int64): [1, 2, 3, 4]
>>> pd.unique(sub_cats)
[1, 2]
Categories (2, int64): [1, 2]
```
Turning it into a `list` does have the side-effect of changing the index from categorical to int, but I don't know if that matters.

Reply:
would `pd.unique(instance_ids).values` work?

Reply:
That gives
```
AttributeError: 'Categorical' object has no attribute 'values'
```

Reply:
weird, it's documented here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Categorical.html#pandas.Categorical

Reply:
using `astype` might be better for the casting.

```
In [14]: l = np.random.choice(["a", "b", "d"], 100000)                                                                                                                                                                                        

In [15]: c = pd.Categorical(l)                                                                                                                                                                                                                

In [16]: %timeit c.astype(pd.Series)                                                                                                                                                                                                          
1.04 ms ± 6.14 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [17]: %timeit c.astype(list)                                                                                                                                                                                                               
1.03 ms ± 9.43 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [18]: %timeit list(c)                                                                                                                                                                                                                      
2.02 ms ± 26.3 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [24]: all(list(c) == c.astype(list))                                                                                                                                                                                                       
Out[24]: True

In [25]: all(list(c) == c.astype(pd.Series))                                                                                                                                                                                                  
Out[25]: True
```

Reply:
We could do `pd.unique(instance_ids).astype(instance_ids.dtype)` which would keep it as `Categorical`. This is also the fastest I think:
```
In [1]: import numpy as np
   ...: import pandas as pd

In [2]: l = np.random.choice(["a", "b", "d"], 100000)
   ...: c = pd.Categorical(l)

In [3]: l2 = np.random.choice(["a", "b"], 100000)
   ...: c2 = pd.Categorical(l2)

In [4]: %timeit c2.astype(list)
1.21 ms ± 81 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [5]: %timeit c2.astype(c.dtype)
529 µs ± 13.4 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```